app/controllers/bills_controller.rb

```diff
@@ -0,0 +1,49 @@
+class BillsController < ApplicationController
+
+  def new
+    begin
+      @session = Stripe::Checkout::Session.create({
+        payment_method_types: ['card'],
+        customer: current_user.customer_id,
+        line_items: get_line_items,
+        mode: 'payment',
+        success_url: books_url,
+        cancel_url: sales_url,
+      })
+    rescue => e
+      Rails.logger.debug e.message
+    end
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  private
+    def sales_ids
+      params.require(:sales)
+    end
+
+    def get_books
+      books = []
+      sales_ids.each do |id|
+        book = Book.find(Sale.find(id).book_id)
```

Comment:
find throws an exception if item not found, you should handle that in your logic